src/components/Filter/FilterHeader.tsx

```diff
@@ -0,0 +1,51 @@
+import { ComponentProps, Dispatch, SetStateAction, forwardRef } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+import { Text, TextVariants } from '../Text';
+import Button from '../Button/Button';
+import { ButtonVariants } from '../Button/types';
+import ChevronNextIcon from '../../assets/chevronNext.svg?react';
+
+export interface FilterHeaderProps extends ComponentProps<'div'> {
+  // Text to display as the header
+  label: string;
+
+  // Whether the show all back button should be displayed (when viewing all filters)
+  showBack?: boolean;
+
+  // Setter function for info to be displayed in View All
+  setViewAllFilter?: Dispatch<SetStateAction<[]>>;
+}
+/**
+ * ## Overview
+ *
+ * The header of a filter
+ */
+const FilterHeader = forwardRef<HTMLDivElement, FilterHeaderProps>(
+  ({ className, label, showBack = false, setViewAllFilter, ...props }, ref) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'FilterHeader');
+
+    return (
+      <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
```

Comment:
should this be a "Legend"? <legend>Choose your favorite monster</legend> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset